File: BaekJoon/Bronze/30802/C++/30802.cpp
// no.30802: 웰컴 키트 (B3)

#include <cstdio>
using namespace std;

int main() {
    int n;
    int s, m, l, xl, xxl, xxxl;
    int t, p;
    scanf("%d", &n);
    scanf("%d %d %d %d %d %d", &s, &m, &l, &xl, &xxl, &xxxl);
    scanf("%d %d", &t, &p);

    int ret1 = 0;
    ret1 += s%t ? s/t+1 : s/t;
    ret1 += m%t ? m/t+1 : m/t;
    ret1 += l%t ? l/t+1 : l/t;
    ret1 += xl%t ? xl/t+1 : xl/t;
    ret1 += xxl%t ? xxl/t+1 : xxl/t;
    ret1 += xxxl%t ? xxxl/t+1 : xxxl/t;
    printf("%d\n", ret1);
    printf("%d %d\n", n/p, n%p);
    return 0;
}

File: BaekJoon/Bronze/30802/Python/30802.py
# no. 30802: 웰컴 키트
# 티셔츠는 같은 사이즈의 T장 묶음으로만 주문 가능
# 펜은 한 종류, P자루씩 묶음으로 주문 or 한자루씩 주문가능
# 사이즈는 S, M, L, XL, XXL, XXXL
# 티셔츠는 남아도 되지만 부족해서는 안되고 펜은 정확하게 줘야함
from math import ceil

n = int(input())
s, m, l, xl, xxl, xxxl = map(int, input().split())
t, p = map(int, input().split())

t_sum = ceil(s/t) + ceil(m/t) + ceil(l/t) + ceil(xl/t) + ceil(xxl/t) + ceil(xxxl/t)
max_p = (s+m+l+xl+xxl+xxxl) // p
cnt_p = (s+m+l+xl+xxl+xxxl) % p

print(t_sum)
print(max_p, cnt_p)